doc/developer-notes.md

```diff
@@ -839,6 +840,37 @@ namespace {
 #endif // BITCOIN_FOO_BAR_H
 ```
 
+Construct On First Use idiom
```

Comment:
perhaps drop "idiom", or more clear as "Construct On First Use" idiom (same for line 853 below)